File: content/project/unit12/cleanNames.js
// cleanNames.js

function processNames(names) {
    // Change: normalize input to lower-case and remove empty entries for stable comparisons
    const normalized = names
        .map((name) => name.trim().toLowerCase())
        .filter(Boolean); // drop blanks after trimming

    // Use Set to deduplicate, then format with first-letter capitalization
    const unique = [...new Set(normalized)].map(
        (name) => name[0].toUpperCase() + name.slice(1)
    );

    // Case-insensitive sort so names appear in predictable order
    unique.sort((a, b) => a.localeCompare(b, undefined, { sensitivity: "base" }));
    return unique;
}

// Test list with duplicates and inconsistent capitalisation
const testNames = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"];

console.log("Original list:", testNames);
console.log("Processed list:", processNames(testNames));

File: content/project/unit12/clean_names.py
# clean_names.py

def process_names(names):
    # Change: normalize to lower-case and drop blanks so duplicates are case-insensitive
    normalized = [name.strip().lower() for name in names if name.strip()]
    # Use set for deduplication, then format with first-letter capitalization
    unique_names = [name.capitalize() for name in set(normalized)]
    # Case-insensitive sort keeps output predictable
    unique_names.sort(key=str.lower)
    return unique_names

# Test list with duplicates and inconsistent capitalisation
test_names = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"]

print("Original list:", test_names)
print("Processed list:", process_names(test_names))
